### PR TITLE
Fixup CI Builld

### PR DESCRIPTION
**Pull request by robbavey**
Add CI scripts for Travis integration
Add gradle wrappers
Update gradle files to build against various branches
Update VERSION and changelog

### CODE REVIEW

**Comment by ph** [comment]
@robbavey 

```groovy

// find sibling logstash directory
def lsDir = null
new File('..').eachDir() { dir ->
    String what = "git --git-dir ${dir.getPath()}/.git remote -v".execute().text
    what.eachLine {
        if (it =~ /elastic.*\/logstash.git/) {
            lsDir = dir
        }
    }
}
```

Do we really need that code? Logstash 5.4.0 shipped with DLQ, so the 5.4.0 released gems on rubygems should have everything need to run the test?

**Comment by robbavey** [comment]
@ph The code has changed since then - the names of the DeadLetterQueueReader and Writer classes were changed between 5.4 and 5.x.

**Comment by ph** [comment]
> The code has changed since then - the names of the DeadLetterQueueReader and Writer classes were changed between 5.4 and 5.x.

Ok,  I've took some time to look at the 5.4 branch, We shipped DLQ related code in 5.4, but we didn't expose the feature to plugin before 5.x

**Comment by suyograo** [comment]
@robbavey one of the unit tests are failing -- can you check? https://travis-ci.org/logstash-plugins/logstash-input-dead_letter_queue/jobs/238862927

**Comment by robbavey** [comment]
@suyograo - this is going to be an interesting one to fix, and maybe broken for a short while, at least for builds that are using the rubygems rather than from a cloned version of the source from a named branch.

- The Java plugin relies on code from the logstash-core jar - I think that this can be fixed using a rake task such as is used in the [logstash dissect filter](https://github.com/logstash-plugins/logstash-filter-dissect/blob/00cd005a3e737fdf4149b15346715d3a28072a5d/Rakefile)
- Also, the plugin relies on code that is only in the 5.x, master and branches derived from those, which have not yet been published to rubygems (other than a 6.0.0-alpha version). 

Going to try and get something working before my training this afternoon

**Comment by robbavey** [comment]
@suyograo 

I think this is in a decent state - Travis is passing against 5.x and master (and should pass against jruby 9000). To work against rubygems, a new version of logstash-core-plugin-api pointing to 5.5+ version of logstash-core will need to be published.

**Comment by suyograo** [comment]
@robbavey this failed against JRuby 9k branch. Looks like a legit issue -- can you take a look please?

**Comment by robbavey** [comment]
Looking @suyograo . That seems strange...

File: Rakefile [hunk 1]
```diff
@@ -3,3 +3,23 @@ require "logstash/devutils/rake"
 task :default do
   system('rake -T')
 end
+
+task :gradle => "gradle.properties" do
+  system("./gradlew vendor")
```
Comment (jakelandis) [inline Rakefile]: Can you apply [this](https://github.com/logstash-plugins/logstash-input-beats/commit/b64c5cbcb0c94edc7147e140caccfb603fa7b677) change here too ? (It protects against releasing a bad gem due to a failure here)

File: Rakefile [hunk 2]
```diff
@@ -3,3 +3,23 @@ require "logstash/devutils/rake"
 task :default do
   system('rake -T')
 end
+
+task :gradle => "gradle.properties" do
```
Comment (jakelandis) [inline Rakefile]: We should have consistent naming for this task.  In [beats input](https://github.com/logstash-plugins/logstash-input-beats/blob/master/lib/tasks/build.rake#L5) it is called ```vendor```. I don't have a naming preference, just a preference for consistency.

Comment (talevy) [reply Rakefile]: more importantly, `vendor` is a first-class citizen in our ci/jarvis-publish setups: https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/master/ci/build.sh#L19

**Comment by robbavey** [comment]
@suyograo Test failure looks like it is the Thread.sleep() waiting for the queue to be filled being too short - I've upped it to see if that fixes the issue

**Comment by suyograo** [comment]
Nice, green now. I'll publish a snapshot for 5.5 branch which should fix one of the red ones.

LGTM

**Comment by robbavey** [comment]
@suyograo Tests are all passing now

**Comment by robbavey** [comment]
@suyograo I missed merging this commit earlier - I'm assuming your (much) earlier LGTM still stands?

**Pull request closed without merging**